patches/minecraft/net/minecraft/entity/monster/EntityZombie.java.patch

```diff
@@ -56,6 +56,15 @@
                              entityzombie.func_180482_a(this.field_70170_p.func_175649_E(new BlockPos(entityzombie)), (IEntityLivingData)null);
                              this.func_110148_a(field_110186_bp).func_111121_a(new AttributeModifier("Zombie reinforcement caller charge", -0.05000000074505806D, 0));
                              entityzombie.func_110148_a(field_110186_bp).func_111121_a(new AttributeModifier("Zombie reinforcement callee charge", -0.05000000074505806D, 0));
+@@ -394,7 +406,7 @@
+             entityzombievillager.func_82149_j(entityvillager);
+             this.field_70170_p.func_72900_e(entityvillager);
+             entityzombievillager.func_180482_a(this.field_70170_p.func_175649_E(new BlockPos(entityzombievillager)), new EntityZombie.GroupData(false));
+-            entityzombievillager.func_190733_a(entityvillager.func_70946_n());
++            entityzombievillager.setForgeProfession(entityvillager.getProfessionForge());
```

Comment:
What I meant is that you should fix getProfession %6 there is no reason for it to be doing that anymore.
Everywhere else supports professions > 6 no reason the getter should be screwing things up.